integration/integration_test.go

```diff
@@ -88,185 +85,134 @@ func init() {
+		s.log = testlog.FailureOnly(t)
+		os.RemoveAll(profile.FullProfilePath(""))
+		t.Cleanup(func() { s.log = nil })
+		test(t, s)
 	}
-	return t
-}
-
-// newTeleportIoT helper returns a running Teleport instance with Host as a
-// reversetunnel node.
-func (s *IntSuite) newTeleportIoT(c *check.C, logins []string) *TeleInstance {
-	// Create a Teleport instance with Auth/Proxy.
-	mainConfig := func() *service.Config {
-		tconf := s.defaultServiceConfig()
-		tconf.Auth.Enabled = true
-
-		tconf.Proxy.Enabled = true
-		tconf.Proxy.DisableWebService = false
-		tconf.Proxy.DisableWebInterface = true
-
-		tconf.SSH.Enabled = false
-
-		return tconf
-	}
-	main := s.newTeleportWithConfig(c, logins, nil, mainConfig())
-
-	// Create a Teleport instance with a Node.
-	nodeConfig := func() *service.Config {
-		tconf := s.defaultServiceConfig()
-		tconf.Hostname = Host
-		tconf.Token = "token"
-		tconf.AuthServers = []utils.NetAddr{
-			{
-				AddrNetwork: "tcp",
-				Addr:        net.JoinHostPort(Loopback, main.GetPortWeb()),
-			},
-		}
-
-		tconf.Auth.Enabled = false
-
-		tconf.Proxy.Enabled = false
-
-		tconf.SSH.Enabled = true
-
-		return tconf
-	}
-	_, err := main.StartReverseTunnelNode(nodeConfig())
-	c.Assert(err, check.IsNil)
-
-	return main
 }
 
 // newTeleportWithConfig is a helper function that will create a running
 // Teleport instance with the passed in user, instance secrets, and Teleport
 // configuration.
-func (s *IntSuite) newTeleportWithConfig(c *check.C, logins []string, instanceSecrets []*InstanceSecrets, teleportConfig *service.Config) *TeleInstance {
-	t := s.newTeleportInstance(c)
+func (s *integrationTestSuite) newTeleportWithConfig(t *testing.T, logins []string, instanceSecrets []*InstanceSecrets, teleportConfig *service.Config) *TeleInstance {
+	teleport := s.newTeleportInstance()
 
 	// use passed logins, but use suite's default login if nothing was passed
 	if len(logins) == 0 {
 		logins = []string{s.me.Username}
 	}
 	for _, login := range logins {
-		t.AddUser(login, []string{login})
+		teleport.AddUser(login, []string{login})
 	}
 
 	// create a new teleport instance with passed in configuration
-	if err := t.CreateEx(instanceSecrets, teleportConfig); err != nil {
-		c.Fatalf("Unexpected response from CreateEx: %v", trace.DebugReport(err))
+	if err := teleport.CreateEx(instanceSecrets, teleportConfig); err != nil {
+		t.Fatalf("Unexpected response from CreateEx: %v", trace.DebugReport(err))
 	}
-	if err := t.Start(); err != nil {
-		c.Fatalf("Unexpected response from Start: %v", trace.DebugReport(err))
+	if err := teleport.Start(); err != nil {
+		t.Fatalf("Unexpected response from Start: %v", trace.DebugReport(err))
 	}
 
-	return t
+	return teleport
 }
 
-// TestAuditOn creates a live session, records a bunch of data through it
+// TestNewStyleIntegrations acts as the master test suite for all integration tests
+// requiring standardised setup and teardown.
+func TestNewStyleIntegrations(t *testing.T) {
```

Comment:
I'm not a big fan of naming things with New - since it will inevitably become old

Reply:
Crud. Detritus from when I had both going at once. Will remove.

Reply:
Done